### PR TITLE
Expand JS support

### PR DESCRIPTION
**Pull request by vast**
We definitely see more and more Javascript in Rails applications. To properly support it we need to address several pain points: JS unit testing, linting and templating.

This PR addresses only JS unit testing (Jasmine) and templating (Skim: Slim + embedded CoffeeScript) since Coffee/Javascript linting requires some investigation.

Addresses:

* #250 
* #249

### CODE REVIEW

**Comment by timurvafin** [comment]
@vast do we need setup phantomjs using autoinstaller instead of `brew`?

File: bin/ci [hunk 1]
```diff
@@ -4,3 +4,4 @@ set -e
 
 bin/rspec spec
 bin/quality
+bin/rake jasmine:ci
```
Comment (timurvafin) [inline bin/ci]: Please run quality tools right after all specs.

**Comment by vast** [comment]
@timurvafin, I think, yes:

1. it will install it only once (not on each `jasmine:ci` run)
2. it doesn't require any additional setup instructions for people not using `brew`, `osx-bootstrap` (customer's developers)

File: bin/ci (hunk 1 above)
Comment (vast) [inline bin/ci]: thanks, fixed in https://github.com/fs/rails-base/commit/037da942558f8dec2ce2a81bc86f2c5d1961b856

**Comment by vast** [comment]
@timurvafin, as we discussed earlier, let's disable automatic installation of PhantomJS. Please have a look at https://github.com/fs/rails-base/commit/97d8395087510ae33f6092a10166fb5743f621c4 and the PR itself.

File: README.md [hunk 2]
```diff
@@ -77,6 +81,10 @@ Run setup script
 
     bin/setup
 
+Install phantomjs 1.9.8 to run Javascript unit tests:
+
+    brew install phantomjs198
```
Comment (timurvafin) [inline README.md]: Please also mention phontamjs in the https://github.com/fs/rails-base/blob/js-support/doc/README_TEMPLATE.md

File: spec/javascripts/support/jasmine_helper.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+#
+# Example: using a different boot file.
+# Jasmine.configure do |config|
+#    config.boot_dir = '/absolute/path/to/boot_dir'
+#    config.boot_files = lambda { ['/absolute/path/to/boot_dir/file.js'] }
+# end
+#
+# Prevent PhantomJS auto install, uses PhantomJS already on your path.
+Jasmine.configure do |config|
+  config.prevent_phantom_js_auto_install = false
```
Comment (timurvafin) [inline spec/javascripts/support/jasmine_helper.rb]: @vast `config.prevent_phantom_js_auto_install` should be set to `true` to disable autoinstall.

Comment (vast) [inline spec/javascripts/support/jasmine_helper.rb]: good catch, thanks. Fixed it in https://github.com/fs/rails-base/commit/e31337a0f9ab8d36c51afeccc2b2b8b2876c5337

File: README.md (hunk 2 above)
Comment (timurvafin) [inline README.md]: @vast ^^^

Comment (vast) [inline README.md]: @timurvafin, done in https://github.com/fs/rails-base/commit/9d44dd644eb52f4ca03ab0cdd39abe8cecb53316

Comment (vast) [inline README.md]: @timurvafin, done in https://github.com/fs/rails-base/commit/9d44dd644eb52f4ca03ab0cdd39abe8cecb53316

**Comment by timurvafin** [comment]
:shipit:

**Pull request merged**